Formula/virtualfish.rb

```diff
@@ -0,0 +1,82 @@
+class Virtualfish < Formula
+  include Language::Python::Virtualenv
+
+  desc "Python virtual environment manager for the fish shell"
+  homepage "https://virtualfish.readthedocs.io/en/latest/"
+  url "https://files.pythonhosted.org/packages/54/2f/a7800ae09a689843b62b3eb423d01a90175be5795b66ac675f6f45349ca9/virtualfish-2.5.5.tar.gz"
+  sha256 "6b654995f151af8fca62646d49a62b5bf646514250f1461df6d42147995a0db2"
+  license "MIT"
+  head "https://github.com/justinmayer/virtualfish.git", branch: "main"
+
+  depends_on "fish"
+  depends_on "python@3.11"
+
+  resource "distlib" do
+    url "https://files.pythonhosted.org/packages/58/07/815476ae605bcc5f95c87a62b95e74a1bce0878bc7a3119bc2bf4178f175/distlib-0.3.6.tar.gz"
+    sha256 "14bad2d9b04d3a36127ac97f30b12a19268f211063d8f8ee4f47108896e11b46"
+  end
+
+  resource "filelock" do
+    url "https://files.pythonhosted.org/packages/24/85/cf4df939cc0a037ebfe18353005e775916faec24dcdbc7a2f6539ad9d943/filelock-3.12.0.tar.gz"
+    sha256 "fc03ae43288c013d2ea83c8597001b1129db351aad9c57fe2409327916b8e718"
+  end
+
+  resource "packaging" do
+    url "https://files.pythonhosted.org/packages/b9/6c/7c6658d258d7971c5eb0d9b69fa9265879ec9a9158031206d47800ae2213/packaging-23.1.tar.gz"
+    sha256 "a392980d2b6cffa644431898be54b0045151319d1e7ec34f0cfed48767dd334f"
+  end
+
+  resource "pkgconfig" do
+    url "https://files.pythonhosted.org/packages/c4/e0/e05fee8b5425db6f83237128742e7e5ef26219b687ab8f0d41ed0422125e/pkgconfig-1.5.5.tar.gz"
+    sha256 "deb4163ef11f75b520d822d9505c1f462761b4309b1bb713d08689759ea8b899"
+  end
+
+  resource "platformdirs" do
+    url "https://files.pythonhosted.org/packages/91/17/3836ffe140abb245726d0e21c5b9b984e2569e7027c20d12e969ec69bd8a/platformdirs-3.5.0.tar.gz"
+    sha256 "7954a68d0ba23558d753f73437c55f89027cf8f5108c19844d4b82e5af396335"
+  end
+
+  resource "psutil" do
+    url "https://files.pythonhosted.org/packages/d6/0f/96b7309212a926c1448366e9ce69b081ea79d63265bde33f11cc9cfc2c07/psutil-5.9.5.tar.gz"
+    sha256 "5410638e4df39c54d957fc51ce03048acd8e6d60abc0f5107af51e5fb566eb3c"
+  end
+
+  resource "virtualenv" do
+    url "https://files.pythonhosted.org/packages/d6/37/3ff25b2ad0d51cfd752dc68ee0ad4387f058a5ceba4d89b47ac478de3f59/virtualenv-20.23.0.tar.gz"
+    sha256 "a85caa554ced0c0afbd0d638e7e2d7b5f92d23478d05d17a76daeac8f279f924"
+  end
+
+  def install
+    virtualenv_install_with_resources
+    system bin/"vf", "install"
+  end
+
+  test do
+    # Pre-create .virtualenvs to avoid interactive prompt
+    (testpath/".virtualenvs").mkpath
+
+    # Run `vf install` in the test environment, adds vf as function
+    refute_path_exists testpath/".config/fish/conf.d/virtualfish-loader.fish"
+    assert_match "VirtualFish is now installed!", shell_output("fish -c '#{bin}/vf install'")
+    assert_path_exists testpath/".config/fish/conf.d/virtualfish-loader.fish"
```

Comment:
If you have to run it here, does that mean the one in the `install` block has no effect?

Reply:
It does have an effect on the user's `fish` config when the user, already using `fish`, runs as `brew install virtualfish`

I think it would work for tests if:

1. Temp test directory set up, set as `$HOME`
2. `fish` runs, sets up `{testdir}/.config/fish` and subfolders
3. `brew install virtualfish` runs with `vf install`

Is there a way to configure `brew test` so that `fish` runs before `brew install virtualfish`?

Reply:
I skimmed the test code, I don't see an easy way to do this.

`virtualfish` is inspired by `virtualenvwrapper`, which works for Posix shells. I was hoping https://github.com/Homebrew/homebrew-core/blob/master/Formula/virtualenvwrapper.rb would have a better strategy, but it also repeats the install in the test:

https://github.com/Homebrew/homebrew-core/blob/88bb13b67f063a1e7032e12c33987eb0ac22ab28/Formula/virtualenvwrapper.rb#L68-L71

I could break up the tests, so that it first just runs `fish` to setup the test directory, rather than combine the first run of `fish` with running `vf install`.

If you run `vf` first, it returns:

```
> vf
VirtualFish is not installed. Run 'vf install' to install it, then run 'exec fish' to restart your shell and load it.
```

When run this way, it is running the Python script, which _only_ looks for the command `install`:

https://github.com/justinmayer/virtualfish/blob/f73ead92ba948e4273c15b22152a5464de871a49/virtualfish/loader/cli.py#L48-L53

When running `vf install`, it installs `fish` functions, so that the next time you run `vf`, you are executing a `fish` function rather than the Python.

Reply:
> It does have an effect on the user's fish config when the user, already using fish, runs as brew install virtualfish

It won't because that only executes when someone builds from source. You'll need a post_install block for that.